### PR TITLE
Cherry-Picking commits from Master needed for 3.5.1

### PR DESCRIPTION
**Pull request by rpmoore**

### CODE REVIEW

File: ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/FileObjectGetter_Test.java [hunk 1]
```diff
@@ -77,4 +85,27 @@ public void testThatFileReportsAsRegularOnWindows() throws IOException, Interrup
 
         assertFalse(caughtException.get());
     }
+
+    @Test
+    public void testThatSymbolicLinksAreResolved() {
+        Assume.assumeTrue(Platform.isMac());
```
Comment (GraciesPadre) [inline ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/FileObjectGetter_Test.java]: Don't we want this to run on linux as well?

Comment (DenverM80) [inline ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/FileObjectGetter_Test.java]: we should cherry-pick this commit as well:
https://github.com/SpectraLogic/ds3_java_sdk/commit/e808e7cd69d41789340a03d936d4aae825155a90

Comment (rpmoore) [inline ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/FileObjectGetter_Test.java]: I'll do it right now

**Comment by rpmoore** [comment]
I added the commit.

**Pull request merged**